src/processor/reconnection-graph.service.ts

```diff
@@ -125,4 +136,179 @@ export class ReconnectionGraphService implements OnApplicationBootstrap, OnAppli
       }
     }
   }
+
+  async formImportBundles(dsnpUserId: MessageSourceId, graphSdkConfig: Config, graphKeyPair: GraphKeyPair): Promise<ImportBundle[]> {
+    const importBundles: ImportBundle[] = [];
+    const public_follow_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Follow, PrivacyType.Public);
+    const public_friendship_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Friendship, PrivacyType.Public);
+    const private_follow_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Follow, PrivacyType.Private);
+    const private_friendship_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Friendship, PrivacyType.Private);
+
+    const publicFollows: PaginatedStorageResponse[] = await this.api.rpc.statefulStorage.getPaginatedStorage(dsnpUserId, public_follow_schema_id);
+    const publicFriendships: PaginatedStorageResponse[] = await this.api.rpc.statefulStorage.getPaginatedStorage(dsnpUserId, public_friendship_schema_id);
+    const privateFollows: PaginatedStorageResponse[] = await this.api.rpc.statefulStorage.getPaginatedStorage(dsnpUserId, private_follow_schema_id);
+    const privateFriendships: PaginatedStorageResponse[] = await this.api.rpc.statefulStorage.getPaginatedStorage(dsnpUserId, private_friendship_schema_id);
+
+    const dsnpKeys = await this.formDsnpKeys(dsnpUserId, graphSdkConfig);
+
+    const importBundleBuilder = new ImportBundleBuilder();
+    // Only X25519 is supported for now
+    const graphKeyType = GraphKeyType.X25519;
+    if (graphKeyPair.keyType !== KeyType.X25519) {
+      throw new Error(`Graph key type ${graphKeyPair.keyType} does not match expected type ${graphKeyType}`);
+    }
+
+    importBundles.push(
+      ...publicFollows.map((publicFollow) => {
+          importBundleBuilder
+            .withDsnpUserId(dsnpUserId.toString())
+            .withSchemaId(public_follow_schema_id)
+            .withPageData(publicFollow.page_id.toNumber(), publicFollow.payload, publicFollow.content_hash.toNumber())
+            .withGraphKeyPair(graphKeyType, graphKeyPair.publicKey, graphKeyPair.privateKey)
```

Comment:
Technically don't need graph keypair for public follows

```suggestion
```

Reply:
yeah its better to not put it to avoid confusion